### PR TITLE
Add support for LMT01 temperature sensor

### PR DESCRIPTION
**Pull request by justifiably**
## Description:

Proposed support for TI's LMT01 two-wire temperature sensor IC.  This IC boasts good accuracy and a wider range compared with other ICs.  It needs a single GPIO to interface.  I have tested with digital GPIO on ESP8266 using a transistor driver as shown in the datasheet.  The chip might also work with only a resistor and the analogue input with a threshold but I haven't tried that.  Also, the datasheet suggests the use of a look-up table to improve the accuracy at the edges of the range, this might be worth adding but would cost a few more bytes.

The code is very simple, I'm happy to put in public domain or remove my moniker from the copyright. 

I've tried to mimic the other drivers, please review.  It compiles against ESP32 but I haven't tested it on one.

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on core ESP8266 V.2.7.1
  - [ ] The code change is tested and works on core ESP32 V.1.12.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by ascillato** [comment]
Hi,

The tests failed due to it is missing the: 

```lua
#define D_SENSOR_LMT01_PULSE   "LMT01 Pulse"
```

in language files.

Please, add it to them.

Thanks.

File: tasmota/language/en_GB.h [hunk 1]
```diff
@@ -681,6 +681,7 @@
 #define D_SENSOR_WINDMETER_SPEED "WindMeter Spd"
 #define D_SENSOR_TELEINFO_RX   "TInfo Rx"
 #define D_SENSOR_TELEINFO_ENABLE "TInfo EN"
+#define D_SENSOR_LMT01_PULSE   "LMT01 Pulse"
```
Comment (ascillato) [inline tasmota/language/en_GB.h:684]: Hi,

This line is missing in the rest of the language files. Please add it to them. Thanks.

**Review by ascillato — CHANGES_REQUESTED** [review]
Hi,

Missing key in language files. Please add it to them. Thanks.

File: tasmota/language/en_GB.h (hunk 1 above)
Comment (justifiably) [reply tasmota/language/en_GB.h:684]: Oops, sorry.  Done now!

**Comment by justifiably** [comment]
Thanks for accepting so swiftly, great!

**Comment by ascillato** [comment]
Thank you for sharing it! :+1:

**Comment by arendst** [comment]
Could you add some documentation to the documentation site. Especially the supported hardware connection. Thank you.

**Comment by ascillato** [comment]
@justifiably Docs are at https://tasmota.github.io/ and if you click on the pencil on the top right of the page, you can edit any page and post a PR. Thanks.

**Comment by justifiably** [comment]
Happy to add some docs.

**Comment by hallard** [comment]
@ascillato the link you gave fire a 404 :-)

**Comment by ascillato** [comment]
@hallard 

Fixed. Thanks!!!

**Pull request merged**